gremlin-core/src/main/java/org/apache/tinkerpop/gremlin/structure/io/gryo/GryoWriter.java

```diff
@@ -131,7 +131,7 @@ public void writeVertexProperty(final OutputStream outputStream, final VertexPro
     public void writeProperty(final OutputStream outputStream, final Property p) throws IOException {
         final Output output = new Output(outputStream);
         writeHeader(output);
-        kryo.writeObject(output, DetachedFactory.detach(p, true));
```

Comment:
cool - something doesn't make sense in that {{GryoWriter}} - very weird looking.